File: Backend/business-server/src/main/java/tn/esprit/entities/Activity.java
package tn.esprit.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.web.bind.annotation.CrossOrigin;

import javax.persistence.*;
import java.util.Date;

@CrossOrigin
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_activity;
    private String name_activity;
    private String descr_activity;

    @Temporal(TemporalType.DATE)
    private Date date_debut_activity;
    @Temporal(TemporalType.DATE)
    private Date date_fin_activity;

    @Enumerated(EnumType.STRING)
    private Type type_activity;
    @Enumerated(EnumType.STRING)
    private Status status_activity= Status.PENDING;
    @Enumerated(EnumType.STRING)
    private Priority priority_activity;


    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "id_objectif")
    private Objectif objectif;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "id_user")
    private User user;

}

File: Backend/Frontend/src/app/models/activity.ts

import { Objectif } from "./objectif";
import { User } from "./user.model";

export class Activity {
    id_activity!: number;
  name_activity!: string;
  descr_activity!: string;
  date_debut_activity!: Date;
  date_fin_activity!: Date;
  type_activity!: string;
  status_activity!: string;
  priority_activity!: string;
  objectif!: Objectif;
  user?: User[];
}
